sdk/python/packages/flet-core/src/flet_core/date_picker.py

```diff
@@ -0,0 +1,342 @@
+from datetime import datetime, date
+from enum import Enum
+from typing import Any, Optional, Union
+
+from flet_core.control import Control, OptionalNumber
+from flet_core.ref import Ref
+from flet_core.text_style import TextStyle
+from flet_core.types import (
+    ResponsiveNumber,
+)
+
+try:
+    from typing import Literal
+except ImportError:
+    from typing_extensions import Literal
+
+KeyboardTypeString = Literal[
+    None,
+    "text",
+    "multiline",
+    "number",
+    "phone",
+    "datetime",
+    "email",
+    "url",
+    "visiblePassword",
+    "name",
+    "streetAddress",
+    "none",
+]
+
+
+class KeyboardType(Enum):
+    NONE = "none"
+    TEXT = "text"
+    MULTILINE = "multiline"
+    NUMBER = "number"
+    PHONE = "phone"
+    DATETIME = "datetime"
+    EMAIL = "email"
+    URL = "url"
+    VISIBLE_PASSWORD = "visiblePassword"
+    NAME = "name"
+    STREET_ADDRESS = "streetAddress"
+
+
+DatePickerModeString = Literal[
+    "day",
+    "year"
+]
+
+
+class DatePickerMode(Enum):
+    DAY = "day"
+    YEAR = "year"
+
+
+DatePickerEntryModeString = Literal[
+    "calendar",
+    "input",
+    "calendarOnly",
+    "inputOnly"
+]
```

Comment:
@FeodorFitsner , I thought https://github.com/flet-dev/flet/pull/697#issuecomment-1345624500 ?

Reply:
Ah, you mean switching to just enums for new controls, right?

Reply:
That's what I thought after reading your comment in that PR.
Even though I will have wished to have both options.

Reply:
so, there is a data class based implementation. But here, sorry, let's keep old-style enums, or I've to rewriteKeyboardTypeString defined in textfield.py. We can rewrite a style later in a separate PR.

Reply:
Why rewrite? why not just import.
something like `from flet.textfield import KeyboardTypeString`
Anyway, ...

Reply:
yep, it's fixed already.

Reply:
@FeodorFitsner, i dont know if you read my comment [here](https://github.com/flet-dev/flet/pull/697#discussion_r1082531040), but these Literals are somehow very important in some cases.
We need to avoid issues like this below:
https://user-images.githubusercontent.com/98978078/213703993-5ddbe02a-0e37-4e4a-a9c6-781e254988d0.png

Reply:
Sure, let's leave literals along enums for now.